### PR TITLE
Support to cache the config file

### PR DESCRIPTION
**Pull request by tsteur**
refs https://github.com/matomo-org/matomo/issues/14367

This supports caching the config ini file. It would be enabled through `$matomoDir/bootstrap.php` by putting this code in there (we cannot use DI or config file for this because they don't exist at the time of the config loading):

```php
<?php

$GLOBALS['ENABLE_CONFIG_PHP_CACHE'] = true;
$GLOBALS['MATOMO_PLUGIN_DIRS'] = array(); // this line is unrelated... just another setting we added recently by using bootstrap.php
```

It would pretty much only work for single server environments... all others would have the problem that the cache invalidation when the config changes wouldn't be synced. Also we do not want to have to document that users need to invalidate the cache when they make changes to it. It would pretty much only help us and I would probably not even document it. When loaded from a regular filesystem, it should be still very fast with the regular ini. If it works nicely, we could document it eventually.

By caching it, it would come from opcache instead of filesystem... and we would not need to do things like all these decodings etc https://github.com/matomo-org/component-ini/blob/master/src/IniReader.php#L149-L384

Any thoughts on this @mattab 

refs DEV-1692

### CODE REVIEW

**Comment by tsteur** [comment]
As discussed we're wanting to give this a try.

File: core/Config/Cache.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        // because the config is not yet loaded we cannot identify the instanceId...
+        // need to use the hostname
+        $dir = $this->makeCacheDir($this->host);
+
+        parent::__construct($dir);
+    }
+
+    private function makeCacheDir($host)
+    {
+        return PIWIK_INCLUDE_PATH . '/tmp/' . $host . '/cache/tracker';
```
Comment (diosmosis) [inline core/Config/Cache.php]: Should we check somewhere if this directory is writable?

Comment (tsteur) [reply core/Config/Cache.php]: @diosmosis the directory might only be created when actually caching a file (the file backend does that). We already check if `PIWIK_INCLUDE_PATH . '/tmp/'` is writable as part of the system check not sure if it's enough?

Comment (diosmosis) [reply core/Config/Cache.php]: That sounds like enough 👍

**Pull request merged**